### ISSUE TITLE
Can't combine startAt(), endAt()…

### ISSUE THREAD

**Issue by oskarrough** [firebase/emberfire#243]
I'm trying to find the last couple of records (tracks from a playlist) using this code:

```js
return this.store.find('track', {
  orderBy: 'playlist',
  equalTo: '-J_somefirebaseid'
});
```

… but I get the following error:

> Error while processing route: channel.index.index Query: Can't combine startAt(), endAt(), and limit(). Use limitToFirst() or limitToLast() instead.

> Error: Query: Can't combine startAt(), endAt(), and limit(). Use limitToFirst() or limitToLast() instead.

Doing `playlist.get('tracks')` works, but I don't want to load ALL those tracks. Only the last ten.

I'm on  Ember 1.11.0, Ember Data 1.0.0-beta.16.1 and EmberFire 1.4.4.

**Comment by tstirrat** [comment]
This is a limitation of the underlying `Firebase` client. EmberFire simply passes the query hash through to  a firebase `Query`.

https://www.firebase.com/docs/web/guide/retrieving-data.html#section-queries

**Comment by oskarrough** [comment]
Aha, thank you. I read through it but couldn't find a way to query what I need. Query a specific model with a equalTo but limit it?

Or since I already have models in a hasMany, is there a way to get only a subset of them?

**Comment by tstirrat** [comment]
Thats a good question. You could get the `hasMany` relationship array and then `.filter` it on the client side. 

But, I know you're really asking about doing it via server call. There isn't a way that I know of off the top of my head.

Even if we got realtime updates via findQuery as requested in #235, this would still not allow `equalTo` + `limitToX` ...

I wonder, for advanced situations if you could create a ref on the `hasMany` key for the model and when a child changes you can update the array manually... something like this (might need some adjustment):

```js
var adapter = store.adapterFor('playlist');
adapter._geRef('playlist').child(model.id).child('tracks')
  .orderByKey()
  .limitToLast(5)
  .on('value', snapshot => {
    var ids = Object.keys(snapshot.val());
    var requests = ids.map((id) => {
      return this.store.find('track', id);
    });
    Ember.RSVP.all(requests).then((tracks) => {
      this.set('last5Tracks', tracks); // might need an Ember.run wrap
    });
  });
```

**Comment by oskarrough** [comment]
@tstirrat I managed to get it working with your example: http://pastie.org/private/sw2kdqoe0jzjvzpwpwmbw

This really, really helps the percieved performance in our app. Thank you! Perhaps this could be packed into a mixin in the adapter or something similar, not sure how it fits.

**Comment by tstirrat** [comment]
glad it's working!

just as a side note, `Object.keys(snapshot.val())` does not give ordering consistent with the server. use `snapshot.forEach`:

```js
let requests = [];
snapshot.forEach((s) => {
  requests.push(this.store.find(s.key()));
});
```

**Comment by tstirrat** [comment]
I have in mind something like this:

```js
model() {
  var playlist = this.modelFor('playlist'); // parent route
  var last5query = playlist.ref() // <-- already possible
    .child('tracks')
      .orderByKey()
      .limitToLast(5);

  return this.store.find('track', { ref: last5query }); // <-- not yet possible
}
```

This would return an `AdapterPopulatedRecordArray` that stays in sync when new items are pushed.

#249 is the first step towards it, by exposing `.ref()`

Allowing query by ref would also allow pagination with [paginate/scroll pseudo refs](https://github.com/firebase/firebase-util/tree/master/src/Paginate)

**Comment by oskarrough** [comment]
@tstirrat with the recent 1.13.0/json-api commits in master, is there now a way for this to happen?

**Comment by tstirrat** [comment]
We still have the limitation in the underlying Firebase library.

As for my last comment. I tried to POC the aforementioned syntax: 

```
this.store.find('track', { ref: someFirebaseQuery });
```

I ran into some problems because the query might return full items (if it's an embedded relationship), or it might return just the `hasMany` relationship data (an object hash of ids). If it only returned the `hasMany` relationship keys it would require a second set of individual lookups to retrieve the actual record payloads.

It definitely could prove useful, but it is not something I am actively looking at, at the moment.
